lib/java_buildpack/framework/dynatrace_one_agent.rb

```diff
@@ -46,83 +46,84 @@ def compile
+        credentials = service['credentials']
 
-        @droplet.java_opts.add_agentpath_with_props(agent_path,
-                                                    SERVER      => server(credentials),
-                                                    TENANT      => tenant(credentials),
-                                                    TENANTTOKEN => tenanttoken(credentials))
+        @droplet.java_opts.add_agentpath(agent_path)
 
         environment           = @application.environment
         environment_variables = @droplet.environment_variables
 
-        unless environment.key?(RUXIT_APPLICATION_ID)
-          environment_variables.add_environment_variable(RUXIT_APPLICATION_ID, application_id)
+        unless environment.key?(DT_APPLICATION_ID)
+          environment_variables.add_environment_variable(DT_APPLICATION_ID, application_id)
         end
 
-        environment_variables.add_environment_variable(RUXIT_HOST_ID, host_id) unless environment.key?(RUXIT_HOST_ID)
+        environment_variables.add_environment_variable(DT_HOST_ID, host_id) unless environment.key?(DT_HOST_ID)
+        environment_variables.add_environment_variable(DT_TENANT, credentials[ENVIRONMENTID])
+        environment_variables.add_environment_variable(DT_TENANTTOKEN, tenanttoken)
+        environment_variables.add_environment_variable(DT_CONNECTION_POINT, endpoints)
       end
 
       protected
 
       # (see JavaBuildpack::Component::VersionedDependencyComponent#supports?)
       def supports?
-        @application.services.one_service? FILTER, [ENVIRONMENTID, TENANT], [APITOKEN, TENANTTOKEN]
-      end
-
-      def supports_apitoken?
-        credentials = @application.services.find_service(FILTER)['credentials']
-        credentials[APITOKEN] ? true : false
+        !service.nil?
       end
 
       private
 
-      FILTER = /ruxit|dynatrace/
+      FILTER = /dynatrace/
 
-      RUXIT_APPLICATION_ID = 'RUXIT_APPLICATIONID'.freeze
+      DT_APPLICATION_ID = 'DT_APPLICATIONID'.freeze
 
-      RUXIT_HOST_ID = 'RUXIT_HOST_ID'.freeze
+      DT_HOST_ID = 'DT_HOST_ID'.freeze
 
-      SERVER = 'server'.freeze
+      DT_TENANT = 'DT_TENANT'.freeze
 
-      TENANT = 'tenant'.freeze
+      DT_TENANTTOKEN = 'DT_TENANTTOKEN'.freeze
 
-      TENANTTOKEN = 'tenanttoken'.freeze
+      DT_CONNECTION_POINT = 'DT_CONNECTION_POINT'.freeze
 
       APITOKEN = 'apitoken'.freeze
 
       APIURL = 'apiurl'.freeze
 
       ENVIRONMENTID = 'environmentid'.freeze
 
-      ENDPOINT = 'endpoint'.freeze
-
-      private_constant :FILTER, :RUXIT_APPLICATION_ID, :RUXIT_HOST_ID, :SERVER, :TENANT, :TENANTTOKEN, :APITOKEN
-      private_constant :ENVIRONMENTID, :ENDPOINT, :APIURL
+      private_constant :FILTER, :DT_APPLICATION_ID, :DT_HOST_ID
+      private_constant :DT_TENANT, :DT_TENANTTOKEN, :DT_CONNECTION_POINT
+      private_constant :ENVIRONMENTID, :APITOKEN
+
+      def service
+        candidates = @application.services.select do |candidate|
```

Comment:
Why did you replicate the `Services.one_service?` method here?  Do you believe that it works differently than the current implementation?

Reply:
The one_service implementation only returns true or false if such service is present.
find_service takes the first service it finds, but it does not take the credentials into account. So that is the difference. It searches for one service with the needed credentials and returns it.